multishard_mutation_query.cc

```diff
@@ -781,6 +782,44 @@ future<foreign_ptr<lw_shared_ptr<typename ResultBuilder::result_type>>> do_query
     co_return f.get0();
 }
 
+template <typename ResultBuilder>
+future<foreign_ptr<lw_shared_ptr<typename ResultBuilder::result_type>>> do_query_tablets(
+        distributed<replica::database>& db,
+        schema_ptr s,
+        const query::read_command& cmd,
+        const dht::partition_range_vector& ranges,
+        tracing::trace_state_ptr trace_state,
+        db::timeout_clock::time_point timeout,
+        noncopyable_function<ResultBuilder()> result_builder_factory) {
+    auto& table = db.local().find_column_family(s);
+    auto erm = table.get_effective_replication_map();
```

Comment:
@tgrabiec do I need to take some tablet-specific lock here, or would `erm` already pin the current tablets for the duration of this read?

Reply:
`erm` will pin everything. But you must hold it captured until you return.

Reply:
Well the coroutine does that for you.

Reply:
I just remembered that when migrating away tablets, we should evict all saved readers that read from that tablet. While the initiator of these reads will not ever come back for them anyway (they will go to the new replica of the tablet), they might hit some use-after-free when being destroyed at the end of their TTL.

Reply:
The erm should be captured by the reader. Though I'm not to happy about using erm on the replica side.